### PR TITLE
testing: update Go to 1.19

### PR DESCRIPTION
**Pull request by dfawley**
RELEASE NOTES: none

### CODE REVIEW

**Review by easwars — DISMISSED** [review]

File: xds/internal/balancer/clusterresolver/configbuilder.go [hunk 1]
```diff
@@ -90,14 +91,17 @@ type priorityConfig struct {
...
+//	└┬──────┬┘
+//	 │      │
+//
 // ┌──────────▼─┐  ┌─▼──────────┐
 // │cluster_impl│  │cluster_impl│
 // └──────┬─────┘  └─────┬──────┘
-//        │              │
+//
+//	│              │
+//
```
Comment (easwars) [inline xds/internal/balancer/clusterresolver/configbuilder.go:104]: ASCII art fiasco.

File: xds/internal/xdsclient/xdsresource/matcher.go [hunk 2]
```diff
@@ -209,17 +209,17 @@ func match(domain, host string) (domainMatchType, bool) {
...
-//  - A match is better if it’s matching pattern type is better
-//    - Exact match > suffix match > prefix match > universal match
-//  - If two matches are of the same pattern type, the longer match is better
-//    - This is to compare the length of the matching pattern, e.g. “*ABCDE” >
-//    “*ABC”
+//   - A match is better if it’s matching pattern type is better
+//   - Exact match > suffix match > prefix match > universal match
+//   - If two matches are of the same pattern type, the longer match is better
+//   - This is to compare the length of the matching pattern, e.g. “*ABCDE” >
+//     “*ABC”
```
Comment (easwars) [inline xds/internal/xdsclient/xdsresource/matcher.go:222]: Sub-bullets

File: security/authorization/engine/engine.go [hunk 3]
```diff
@@ -255,11 +255,16 @@ func getDecision(engine *policyEngine, match bool) Decision {
 
 // Returns the authorization decision of a single policy engine based on activation.
 // If any policy matches, the decision matches the engine's action, and the first
-//  matching policy name will be returned.
+//
```
[the remark this answers is not in the document]
Comment (dfawley) [reply security/authorization/engine/engine.go:259]: My guess is it was the weird spacing in place before.

File: status/status.go [hunk 4]
```diff
@@ -76,14 +76,14 @@ func FromProto(s *spb.Status) *Status {
...
-//   *Status`, the appropriate Status is returned.
+//   - If err was produced by this package or implements the method `GRPCStatus()
+//     *Status`, the appropriate Status is returned.
 //
 // - If err is nil, a Status is returned with codes.OK and no message.
 //
-// - Otherwise, err is an error not compatible with this package.  In this
-//   case, a Status is returned with codes.Unknown and err's Error() message,
-//   and ok is false.
+//   - Otherwise, err is an error not compatible with this package.  In this
```
[the remark this answers is not in the document]
Comment (dfawley) [reply status/status.go:84]: Fixed (I guess? they all look like sub-bullets now)

File: xds/internal/balancer/clusterresolver/configbuilder.go (hunk 1 above)
Comment (dfawley) [reply xds/internal/balancer/clusterresolver/configbuilder.go]: Fixed?

File: xds/internal/balancer/priority/balancer_priority.go [hunk 5]
```diff
@@ -53,18 +53,18 @@ var (
...
+//   - If this is the lowest priority
 //   - do the following:
-//     - if this is not the old childInUse, override picker so old picker is no
-//       longer used.
-//     - switch to it (because all higher priorities are neither new or Ready)
-//     - forward the new addresses and config
+//   - if this is not the old childInUse, override picker so old picker is no
+//     longer used.
+//   - switch to it (because all higher priorities are neither new or Ready)
+//   - forward the new addresses and config
```
[the remark this answers is not in the document]
Comment (dfawley) [reply xds/internal/balancer/priority/balancer_priority.go]: Fixed as a code block

File: xds/internal/xdsclient/xdsresource/matcher.go (hunk 2 above)
Comment (dfawley) [reply xds/internal/xdsclient/xdsresource/matcher.go]: Fixed??

File: balancer/rls/config.go [hunk 6]
```diff
@@ -116,28 +116,31 @@ type lbConfigJSON struct {
...
+//   - must not have two entries with the same `Name`
+//   - within each entry:
+//   - must have at least one `Name`
+//   - must not have a `Name` with the `service` field unset or empty
+//   - within each `headers` entry:
+//   - must not have `required_match` set
+//   - must not have `key` unset or empty
+//   - across all `headers`, `constant_keys` and `extra_keys` fields:
+//   - must not have the same `key` specified twice
+//   - no `key` must be the empty string
```
[the remark this answers is not in the document]
Comment (dfawley) [reply balancer/rls/config.go:129]: Missed this.  It looks really bad without a marker at the start of the line.  I used `/`.

File: balancer/grpclb/grpclb.go [hunk 7]
```diff
@@ -291,13 +293,13 @@ func (lb *lbBalancer) regeneratePicker(resetDrop bool) {
...
-//    they start to connect immediately. But there's a race between the overall
-//    state is reported, and when the new SubConn state arrives. And SubConns
-//    never go back to IDLE.
-//  - Else the aggregated state is TransientFailure.
+//   - If at least one SubConn in Ready, the aggregated state is Ready;
+//   - Else if at least one SubConn in Connecting or IDLE, the aggregated state is Connecting;
+//   - It's OK to consider IDLE as Connecting. SubConns never stay in IDLE,
+//     they start to connect immediately. But there's a race between the overall
+//     state is reported, and when the new SubConn state arrives. And SubConns
+//     never go back to IDLE.
```
Comment (easwars) [inline balancer/grpclb/grpclb.go:301]: This one was a sub-bullet earlier. Sorry, missed this in the last pass.

File: internal/balancergroup/balancergroup.go [hunk 8]
```diff
@@ -186,18 +186,18 @@ func (sbc *subBalancerWrapper) stopBalancer() {
...
 //
 // Updates from ClientConn are forwarded to sub-balancers
-//  - service config update
-//  - address update
-//  - subConn state change
-//     - find the corresponding balancer and forward
+//   - service config update
+//   - address update
+//   - subConn state change
+//   - find the corresponding balancer and forward
```
Comment (easwars) [inline internal/balancergroup/balancergroup.go:192]: This was sub-bullet

File: internal/balancergroup/balancergroup.go [hunk 9]
```diff
@@ -186,18 +186,18 @@ func (sbc *subBalancerWrapper) stopBalancer() {
...
 //
 // Actions from sub-balances are forwarded to parent ClientConn
-//  - new/remove SubConn
-//  - picker update and health states change
-//     - sub-pickers are sent to an aggregator provided by the parent, which
+//   - new/remove SubConn
+//   - picker update and health states change
+//   - sub-pickers are sent to an aggregator provided by the parent, which
```
Comment (easwars) [inline internal/balancergroup/balancergroup.go:197]: This was a sub-bullet

File: xds/internal/balancer/ringhash/ringhash.go [hunk 10]
```diff
@@ -327,10 +327,11 @@ func (b *ringhashBalancer) ResolverError(err error) {
 // It triggers an update to cc when:
 // - the new state is TransientFailure, to update the error message
 //   - it's possible that this is a noop, but sending an extra update is easier
-//   than comparing errors
+//     than comparing errors
+//
 // - the aggregated state is changed
 //   - the same picker will be sent again, but this update may trigger a re-pick
-//   for some RPCs.
+//     for some RPCs.
```
Comment (easwars) [inline xds/internal/balancer/ringhash/ringhash.go:334]: Interesting that gofmt didnt reformat this as single list. Should we just rewrite this comment as:

```golang
// UpdateSubConnState updates the per-SubConn state stored in the ring, and also
// the aggregated state.
//
// It triggers an update to cc when:
// - the new state is TransientFailure to update the error message. 
//   It's possible that this is a noop, but sending an extra update is easier
//   than comparing errors
// - the aggregated state has changed. The same picker will be sent again, 
//   but this update may trigger a re-pick for some RPCs.
```

File: balancer/rls/config.go [hunk 11]
```diff
@@ -114,30 +114,34 @@ type lbConfigJSON struct {
...
+//	  value is greater than 5M, we cap it at 5M
+//
 // - routeLookupChannelServiceConfig:
 //   - if specified, must parse as valid service config
+//
 // - childPolicy:
 //   - must find a valid child policy with a valid config
+//
 // - childPolicyConfigTargetFieldName:
 //   - must be set and non-empty
```
Comment (easwars) [inline balancer/rls/config.go:146]: Should we just reformat this as a code block. The `/` make the whole thing harder to read. And it does not keep `routeLookupConfig`, `routeLookupChannelServiceConfig`, `childPolicy` and `childPolicyConfigTargetFieldName` at the same indent level.

Comment (dfawley) [reply balancer/rls/config.go:146]: This _is_ a code block; the slash prevents it from being detected as a list instead.  But I found another workaround which I will try here and elsewhere.

File: balancer/rls/config.go [hunk 12]
```diff
@@ -113,33 +113,36 @@ type lbConfigJSON struct {
...
+//	       - across all `headers`, `constant_keys` and `extra_keys` fields:
+//	         - must not have the same `key` specified twice
+//	         - no `key` must be the empty string
+//	- `lookup_service` field must be set and must parse as a target URI
+//	- if `max_age` > 5m, it should be set to 5 minutes
+//	- if `stale_age` > `max_age`, ignore it
+//	- if `stale_age` is set, then `max_age` must also be set
+//	- ignore `valid_targets` field
+//	- `cache_size_bytes` field must have a value greater than 0, and if its
+//	  value is greater than 5M, we cap it at 5M
```
Comment (easwars) [inline balancer/rls/config.go:136]: Sorry, these should be sub-bullets inside of `routeLookupConfig`. They are fields in the route lookup config. This should be the last one.

**Review by easwars — APPROVED** [review]

File: balancer/rls/config.go (hunk 12 above)
Comment (dfawley) [reply balancer/rls/config.go]: Whoops; my mistake.  Fixed

**Review by easwars — APPROVED** [review]

**Pull request merged**